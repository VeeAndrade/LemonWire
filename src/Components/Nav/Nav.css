/* #00aa4e green*/
/* #fdf200 yellow*/
/* #292c33 charcoal */

* {
  margin: 0px;
  padding: 0px;
}

#title {
  position: absolute;
  top: 30%;
  left: 7%;
  font-size: 160%;
  font-family: Arial;
}

#nav {
  height: 80px;
  background-image: url(../../Assets/nav_background.png);
  background-size: contain;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  top: 0;
  left: 0;
  position: fixed;
  z-index: 5;

}

#logo {
  background-image: url(../../Assets/lemonwire_logo.jpg);
  background-size: contain;
  height: 80px;
  width: 80px;
  border-radius: 40px;
}

#logout-button {
  height: 30px;
  width: 80px;
  background-color: #00aa4e;
  border-radius: 15px;
  color: #FFF;
  font-size: 80%;
  border: none;
}

#logout-button:hover {
  border: none;
  cursor: pointer;
  background-color: #fdf200;
  color: #000;
}
