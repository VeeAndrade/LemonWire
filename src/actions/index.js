export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const updateUser = user => ({
  type: 'UPDATE_USER',
  user
});

export const updateSelectedMovie = movie => ({
  type: 'UPDATE_SELECTED_MOVIE',
  movie
});

export const logoutUser = user => ({
  type: 'LOGOUT_USER',
  user
})

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
})

export const removeRatings = ratings => ({
  type: 'REMOVE_RATINGS',
  ratings
})
