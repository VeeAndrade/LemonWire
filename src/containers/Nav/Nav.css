/* #fbe832 yellow*/
/* #292c33 charcoal */

* {
  margin: 0px;
  padding: 0px;
  font-family: Varela Round;
}

#title {
  font-size: 250%;
  font-family: Fredoka One;
  font-weight: lighter;
  position: absolute;
  left: 80px;
  margin-left: 5px;
  color: #FFF;
}

.nav {
  align-items: center;
  background-image: url(../../Assets/film_reel_nav_yellow_nf.png);
  background-size: contain;
  border-top-left-radius: 40px;
  border-bottom-left-radius: 40px;
  display: flex;
  height: 80px;
  justify-content: space-between;
  left: 0;
  position: fixed;
  opacity: 90%;
  top: 0;
  width: 100%;
  z-index: 120;
}

#logo {
  background-image: url(../../Assets/lemon_slice.jpg);
  background-size: contain;
  height: 80px;
  width: 80px;
  border-radius: 40px;
}

.login-button,
.logout-button {
  border: none;
  color: #fbe832;
  font-size: 100%;
  font-weight: bold;
  height: 40px;
  text-decoration: none;
  width: 80px;
}

.login-button:hover,
.logout-button:hover {
  background-color: #fbe832;
  border: none;
  color: #292c33;
  cursor: pointer;
}

.login-button {
  height: 36px !important;
  padding-top: 13px;
  text-align: center;
  margin-top: -1px;
}

#welcome-message {
  color: #FFF;
  margin-left: 20%;
}

#email {
  background-color: #a3ffcd;
  font-size: 80%;
}
