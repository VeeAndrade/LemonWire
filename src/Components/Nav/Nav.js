import React, { Component } from 'react';
import './Nav.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../../actions'

class Nav extends Component {

  render() {
    return (
      <nav className='nav'>
        <div id='logo'></div>
        <h1 id='title'>LemonWire</h1>
        <h2 id='welcome-message'>üçã Welcome {this.props.user ? this.props.user.name : ''}! üçã</h2>
        <div id='user-avatar'></div>
        {this.props.user.id ? <button onClick={() => this.props.logoutUser(this.props.user)} className='logout-button'>Logout</button> : <Link to="/login" className='login-button'>Login</Link>}
      </nav>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user
})

const mapDispatchToProps = dispatch => ({
  logoutUser: user => dispatch(logoutUser(user))
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
