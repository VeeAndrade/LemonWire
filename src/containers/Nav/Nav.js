import React, { Component } from 'react';
import './Nav.css';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { logoutUser } from '../../actions';
import { removeRatings } from '../../actions';

export class Nav extends Component {

  render() {
    return (
      <nav className='nav'>
        <div id='logo'></div>
        <h1 id='title'>LemonWire</h1>
        <h2 id='welcome-message'>{this.props.user.id ? `Welcome, ${this.props.user.name}!` : 'login to rate movies!'}</h2>
        <div id='user-avatar'></div>
        {this.props.user.id ? <button onClick={() => { this.props.logoutUser(this.props.user) && this.props.removeRatings(this.props.ratings)}} className='logout-button'>Logout</button> : <Link to="/login" className='login-button'>Login</Link>}
      </nav>
    )
  }
}

export const mapStateToProps = state => ({
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  logoutUser: user => dispatch(logoutUser(user)),
  removeRatings: ratings => dispatch(removeRatings(ratings))
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
