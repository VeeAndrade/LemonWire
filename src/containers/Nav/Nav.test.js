import React from 'react';
import ReactDOM from 'react-dom';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav';
import { shallow } from 'enzyme';
import { logoutUser } from '../../actions'
import { removeRatings } from '../../actions'

describe('Nav', () => {
  describe('Nav container/component', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Nav user={{id: 5}}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with a user', () => {
      const mockState = {
        movies: [ { name: 'Jumanji' } ],
        user: { name: 'Robbie', id: 5 },
        ratings: [{rating: 5}, {rating: 4}],
      };
      const expected = {
        user: { name: 'Robbie', id: 5 },
      }
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected)
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the logoutUser action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const mockUser = { user: { name: 'Robbie', id: 5 } };
      const actionToDispatch = logoutUser(mockUser);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.logoutUser(mockUser)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch with the removeRatings action when the logout button is clicked', () => {
      const mockDispatch = jest.fn();
      const mockRatings = { ratings: [{rating: 5}, {rating: 4}] };
      const actionToDispatch = removeRatings(mockRatings);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.removeRatings(mockRatings)
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
});
