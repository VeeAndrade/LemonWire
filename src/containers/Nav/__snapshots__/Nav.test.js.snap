// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render a Nav 1`] = `undefined`;
